### ISSUE TITLE
Unable to merge or otherwise fix a duplicate variant for a gene

### ISSUE THREAD

**Issue by malachig** [genome/civic-server#88]
While Cody was adding coordinates for variants he noted that two variants had been created for KRAS, one called 'EXON 1 MUTATIONS' the other called 'EXON 2 MUTATIONS'.  

'EXON 1 MUTATIONS' is a labeling ambiguity and is actually referring to the same mutations as 'EXON 2 MUTATIONS'.  

Cody renamed the exon 1 variant to exon 2.  But of course merging is not that simple as these are just names not the actual variant IDs.  Should such a rename have been allowed?  Anyway, to fix this ...

I would like to move this evidence item:
https://civic.genome.wustl.edu/#/events/genes/30/summary/variants/74/summary/evidence/250/summary

From this variant:
https://civic.genome.wustl.edu/#/events/genes/30/summary/variants/74/summary

To this variant:
https://civic.genome.wustl.edu/#/events/genes/30/summary/variants/75/summary

And then delete variant `74`.  There are various barriers to doing so.  For one thing, in the admin interface:
- It would be helpful if you could filter down to a specific evidence item, in this case `250`
- If you filter to KRAS, this helps but now you have three evidence items that now say `EXON 2 MUTATIONS`, two of which are associated with variant `75` and one with variant `74`.  But you can't see this in the interface directly.
- If I figure out which is associated with variant `74` and edit it I see this:
 
https://civic.genome.wustl.edu/admin/evidence_items/250/edit

- In this view I can not change the evidence item to variant `74` because I only have a drop down list of variant names that are non-unique.  `EXON 2 MUTATIONS` appears three times in this list.  `EXPRESSION` appears 9 times.

My next strategy was to attempt to just create an entirely new evidence item that duplicated the content of the one I want to merge and then just associate that with the correct variant `75`.  That attempt failed and can be seen here:
https://civic.genome.wustl.edu/admin/evidence_items/530

I guess that is related to the issue @obigriffith reported recently.  This is a reminder to clean up these KRAS mutations when we have a way to do this either in the admin interface or civic-client interface.

In general I think we should talk about how we want to handle duplicate variant names within a gene and across genes.  As a starting point we could disallow the former? There may be places where we want to scope variant queries to within the gene. 

Finally, in general in the admin interface I think it would be helpful to display the unique civic database IDs for variants, genes, evidence items, etc.

**Comment by bainscou** [comment]
This is a really good point. It would be really useful to move evidence statements around somehow. It is feasible that variants like the referenced ‘EXON 1 MUTATIONS’ or others like ‘EXON 2 DELETIONS’ could eventually become a variant group as we refine our understanding of the specific lesions in these genes/exons that relate to the clinical action identified.

**Comment by malachig** [comment]
We also have this problem of variants that need to be merged here:
https://civic.genome.wustl.edu/#/events/genes/45/summary/variants/123/summary

We have two variants for TP53 that are both `R249`.  We need to move this evidence item:
https://civic.genome.wustl.edu/#/events/genes/45/summary/variants/123/summary/evidence/399/summary

Onto this variant:
https://civic.genome.wustl.edu/#/events/genes/45/summary/variants/119/summary

And then delete this redundant variant:
https://civic.genome.wustl.edu/#/events/genes/45/summary/variants/123/summary

**Comment by kkrysiak** [comment]
We have another set of variants to be merged. Cody renamed one of them as it was redundant which I agreed with and accepted the change. In retrospect, I may have jumped the gun since this problem hasn't been resolved. However, just 1 evidence statement needs to be moved over.

https://civic.genome.wustl.edu/#/events/genes/15/summary/variants/188/summary

**Comment by malachig** [comment]
@kkrysiak that is okay, lets just keep making note of them here until we have a way to complete the merges.  I believe the plan is for @acoffman to create a path for us in the admin interface to do this.

**Comment by acoffman** [comment]
Yeah, I'm working on a general "utilities" page for the admin that lets you guys do tasks that don't fit neatly into the "big table of evidence items/variants/genes" that the other tabs present. Among them are: merging users, merging variants, and pulling in new sources from pubmed.

**Comment by kkrysiak** [comment]
Another gene with some variants to be cleaned up is EGFR. There are currently "Amplification", "Copy number alteration" and "Increased copy number" as variants, each with 1-2 associated evidence statements.

https://civic.genome.wustl.edu/#/events/genes/19/summary/variants/191/summary/evidence/472/summary

**Comment by kkrysiak** [comment]
Another variant to tackle here: FGFR3-TACC3 associated with FGFR3 should be FGFR2-TACC3 and associated with FGFR2. 

https://civic.genome.wustl.edu/#/events/genes/23/summary/variants/54/summary/evidence/296/summary

**Comment by kkrysiak** [comment]
I've addressed the KRAS (variant 74), TP53 (variant 123) and CEBPA (variant 188) variants.

We tried to address the FGFR3-TACC3 but problems with the merge led to opening issue #110.

The EGFR variants are being addressed by https://github.com/genome/civic-client/issues/233 unless we choose to rename and merge them as discussed in the issue.

@acoffman, unless you want to re-open this to remind you of additional utilities, for the reasons above, I am closing this issue.
